File: semana 5/subida de archivos flask - Portafolio/models/categoria.py
from sqlalchemy.sql.expression import null
from config.base_datos import bd
from sqlalchemy import Column, types
from sqlalchemy.orm import relationship
class CategoriaModel(bd.Model):
  __tablename__ ="t_categoria"
  categoriaId = Column(
    name= 'cat_id',
    type_= types.Integer,
    primary_key=True,
    autoincrement=True,
    unique=True,
    nullable=True
  )

  categoriaNombre=Column(
    name='cat_nombre',
    type_=types.String(45),
    nullable=False
  )

  categoriaOrden=Column(
    name='cat_orden',
    type_=types.Integer,
    nullable=False
  )

  categoriaEstado=Column(
    name='cat_estado',
    type_=types.Boolean(),
    default=True,
    nullable=False
  )

  conocimientos = relationship('ConocimientoModel', backref='categoriaConocimientos', cascade='all, delete')

  def __init__(self, nombre, orden, estado):
    self.categoriaNombre = nombre,
    self.categoriaOrden = orden
    if estado:
      self.categoriaEstado=estado

  def save(self):
    bd.session.add(self)
    bd.session.commit()

      

File: semana 7/CRUD_SEQUALIZE_EXPRESS/src/models/Categoria.js
const { DataTypes } = require("sequelize");
const { conexion } = require("../config/Sequelize");

module.exports = categoria_model = () => {
  return (categoria = conexion.define(
    "categoria",
    {
      categoriaId: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        allowNull: false,
        autoIncremente: true,
        field: "cat_id",
      },
      categoriaNombre: {
        type: DataTypes.STRING(45),
        unique: true,
        allowNull: false,
        field: "cat_nombre",
      },
    },
    {
      tableName: "t_categoria",
      timestamps: false,
    }
  ));
};
